tests/store/artifact/test_models_artifact_repo.py

```diff
@@ -90,11 +109,44 @@ def test_models_artifact_repo_init_with_version_uri_and_db_profile():
         mock_repo.assert_called_once_with(final_uri)
 
 
+@pytest.mark.parametrize(
+    "artifact_location", ["dbfs:/databricks/mlflow-registry/12345/models/keras-model"]
+)
+def test_models_artifact_repo_init_with_version_uri_and_db_profile_from_context(
+    mock_get_model_version_download_uri,
+):  # pylint: disable=unused-argument
+    model_uri = "models:/MyModel/12"
+    with mock.patch(
+        "mlflow.store.artifact.dbfs_artifact_repo.DbfsRestArtifactRepository", autospec=True
+    ) as mock_repo, mock.patch("mlflow.get_registry_uri", return_value="databricks://scope:key"):
+        models_repo = ModelsArtifactRepository(model_uri)
+        assert models_repo.artifact_uri == model_uri
+        assert isinstance(models_repo.repo, DbfsRestArtifactRepository)
+        mock_repo.assert_called_once_with(
+            "dbfs://scope:key@databricks/databricks/mlflow-registry/12345/models/keras-model"
+        )
+
+
+@pytest.mark.parametrize(
+    "artifact_location", ["dbfs:/databricks/mlflow-registry/12345/models/keras-model"]
+)
+def test_models_artifact_repo_init_with_version_uri_and_bad_db_profile_from_context(
+    mock_get_model_version_download_uri,
+):  # pylint: disable=unused-argument
+    model_uri = "models:/MyModel/12"
+    with mock.patch("mlflow.get_registry_uri", return_value="databricks://scope:key:invalid"):
+        with pytest.raises(MlflowException) as ex:
+            ModelsArtifactRepository(model_uri)
+        assert "Key prefixes cannot contain" in ex.value.message
```

Comment:
@ankitmathur-db is something like this what you were thinking? please do let me know if there are other cases you were thinking of I didn't cover here. thanks!

Reply:
Yeah this is what I was thinking. Have we tested stuff like `databricks://scope:key.  ` (with the extra spaces at the end)?

Reply:
Ah we don't test it - but that should be handled at a lower level. Created a PR to address that: https://github.com/mlflow/mlflow/pull/3338/commits/00cab634f5b6bf310310442e66787823abde3418 - will merge this PR and rebase to clean up that PR's commit history.